### ISSUE TITLE
Silent exit when first used in a thread other than the main thread

### ISSUE THREAD

**Issue by allquixotic** [xml4r/libxml-ruby#94]
On Windows 7 32-bit, Ruby 1.9.3-p545 (obtained from RubyInstaller.org).

Using the `libxml-ruby` gem version 2.7.0 for the `x86-mingw32` platform (I installed the binaries, not compiled from source), obtained from rubygems.org.

------

The following program works fine:

```
require 'libxml'
z = LibXML::XML::Parser.string('<?xml version="1.0" encoding="UTF-8"?><root></root>').parse()
worker = Thread.new { LibXML::XML::Parser.string('<?xml version="1.0" encoding="UTF-8"?><root></root>').parse() }
worker.join
puts "Done successfully"
```

The following program prints nothing (it exits silently, no exception raised):

```
require 'libxml'
worker = Thread.new { LibXML::XML::Parser.string('<?xml version="1.0" encoding="UTF-8"?><root></root>').parse() }
worker.join
puts "Done successfully"
```

The difference is that in the second program, the first use of LibXML is on a thread other than the main thread, whereas in the first program, the first use of LibXML is on the main thread.

If the library must do some initialization on the main thread, this should be rolled into the code that gets executed when you run `require`. It should not be necessary to parse a trivial XML document on the main thread in order to be able to use it in a child thread.

Furthermore, this problem is particularly egregious, because it **silently** terminates the process, without giving the programmer any indication of what is wrong. Even a segmentation fault and the accompanying stack dump provides much more information than your process exiting without a peep. Nothing to stdout, nothing to stderr, no exceptions thrown in Ruby, no worrisome signals being emitted by the process to the calling shell, nothing. This can make the error extremely difficult to detect, necessitating a careful step through of all Ruby code to find this error. Please see if you can handle this error in some way other than calling `exit()`.

**Comment by cfis** [comment]
This is fixed in bd529e995f0ccd29f10f6067760bc55f738065c2.

**Comment by cfis** [comment]
Also see #58.
